### ISSUE TITLE
Indexing Variable and Using CPI Offset causing errors

### ISSUE THREAD

**Issue by andersonfrailey** [PSLmodels/Tax-Calculator#2363]
Some Tax-Brain runs have been failing when a user attempts to change `CPI_offset` and toggle the `indexed` variable on another parameter. I have been able to recreate this error when working directly with taxcalc:

```
In [11]: pol = tc.Policy()                                                                                                                                                                                                                                                      

In [12]: reform = {"CTC_c-indexed": {2020: True}}                                                                                                                                                                                                                               

In [13]: pol.implement_reform(reform)                                                                                                                                                                                                                                           

In [14]: pol2 = tc.Policy()                                                                                                                                                                                                                                                     

In [15]: reform2 = {"CPI_offset": {2020: -0.0030}}                                                                                                                                                                                                                              

In [16]: pol2.implement_reform(reform2)                                                                                                                                                                                                                                         

In [17]: pol3 = tc.Policy()                                                                                                                                                                                                                                                     

In [18]: reform3 = {"CPI_offset": {2020: -0.0030}, 
    ...:            "CTC_c-indexed": {2020: True}}                                                                                                                                                                                                                              

In [19]: pol3.implement_reform(reform3)                                                                                                                                                                                                                                         
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-19-9fca5d29e042> in <module>
----> 1 pol3.implement_reform(reform3)

~/anaconda/envs/taxbrain-dev/lib/python3.6/site-packages/taxcalc/policy.py in implement_reform(self, reform, print_warnings, raise_errors)
    126         the expected structure of the reform dictionary.
    127         """
--> 128         self._update(reform, print_warnings, raise_errors)
    129 
    130     @staticmethod

~/anaconda/envs/taxbrain-dev/lib/python3.6/site-packages/taxcalc/parameters.py in _update(self, revision_, print_warnings, raise_errors)
    333             raise ValueError(self.parameter_errors)
    334         # optionally apply CPI_offset to inflation_rates and re-initialize
--> 335         known_years = self._apply_cpi_offset_in_revision(revision)
    336         if known_years is not None:
    337             self._set_default_vals(known_years=known_years)

~/anaconda/envs/taxbrain-dev/lib/python3.6/site-packages/taxcalc/parameters.py in _apply_cpi_offset_in_revision(self, revision)
    848         for year in sorted(revision.keys()):
    849             for name in revision[year]:
--> 850                 if self._vals[name]['indexed']:
    851                     if name not in known_years:
    852                         known_years[name] = kyrs_in_revision

KeyError: '_CTC_c-indexed'
```

As you can see, there's no issue when including just `CTC_c-indexed` or just `CPI_offset` in a reform, but including them together causes an error.

I believe this is because, when applying the CPI offset, taxcalc loops through each parameter in the reform and attempts to access its value in `self._vals`, but no parameters in `self._vals` would end with `-indexed`. A solution would be to skip any parameter in the reform that ends with `-indexed`. Or taxcalc could remove `-indexed` from the parameter name before attempting to access the associated item.

I'm happy to work on this, I just wanted to open an issue first to be sure I wasn't missing anything.

cc @hdoupe @martinholmer

**Comment by MattHJensen** [comment]
I imagine Martin will want to offer feedback on this when he is back from vacation, and it won't make sense to have another Tax-Calculator release before then anyways. So what do you think about proceeding however you think is most wise in developing a PR, and just sharing the understanding that Martin and I might choose to pursue a different direction after we both have had an opportunity to look at your PR and think about it?

**Comment by MattHJensen** [comment]
Also, @andersonfrailey, thank you very much for alerting the project to this issue!

**Comment by andersonfrailey** [comment]
@MattHJensen working on something now sounds good with me. I'll get a PR up today or tomorrow.

**Comment by MattHJensen** [comment]
Thanks, Anderson. Much appreciated.

**Comment by martinholmer** [comment]
This bug report has been resolved by pull request #2364.
